pkg/controller/publishingstrategy/publishingstrategy_controller.go

```diff
@@ -0,0 +1,217 @@
+// and what is in the PublishingStrategy.Spec
+// TODO(user): Modify this Reconcile function to implement your Controller logic.  This example creates
+// a Pod as an example
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcilePublishingStrategy) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling PublishingStrategy")
+
+	// Fetch the PublishingStrategy instance
+	instance := &cloudingressv1alpha1.PublishingStrategy{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// Reconcile will handle
+	// 1. Set the cluster API to Internal
+	// 2. Set the cluster API to External (Internet-facing)
+	// 3. Set the default ingress (application) to Internal
+	// 4. Set the default ingress (application) to External (Internet-facing)
+
+	// get region
+	region, err := utils.GetClusterRegion(r.client)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	// Secret should exist in the same namespace Account CR's are created
+	awsClient, err := awsclient.GetAWSClient(r.client, awsclient.NewAwsClientInput{
+		SecretName: config.AWSSecretName,
+		NameSpace: 	config.OperatorNamespace,
+		AwsRegion: 	region
+	})
+	if err != nil {
+		reqLogger.Error(err, "Failed to get AWS client")
+		return reconcile.Result{}, err
+	}
+
+	// To make API server non-public
+	// 1. Delete ext NLB for port 6443/TCP
+	// 2. Update api.<cluster-domain> DNS record to point to internal NLB
+	if instance.Spec.DefaultAPIServerIngress.Listening == cloudingressv1alpha1.Internal {
+		// get a list of load balancers in use
+		descLBInput := elbv2.DescribeLoadBalancersInput{}
+		descLBOutput, err := awsClient.DescribeLoadBalancersV2(&descLBInput)
+		if err != nil {
+			log.Error(err, "Error getting list of NLBs")
+			return reconcile.Result{}, err
+		}
+		
+		// get the internet-facing NLB by scheme and delete
+		var internalNLBDNS *string
+		var internalCanonicalHostedZoneID *string
+		for _, loadBalancers := range descLBOutput.LoadBalancers{
+			external := "internet-facing"
+			scheme := *loadBalancers.Scheme
+			if scheme == external {
+				//need to get Amazon Resource Name (ARN) of the load balancer to pass in input
+				extARN := loadBalancers.LoadBalancerArn
+				deleteLBInput := elbv2.DeleteLoadBalancerInput{extARN}
```

Comment:
Can you combine these? So it's `elbv2.DeleteLoadBalancerInput{loadBalancers.LoadBalancerArn}`?

Reply:
yes